app/src/test/java/org/mozilla/fenix/gleanplumb/NimbusMessagingControllerTest.kt

```diff
@@ -0,0 +1,173 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.gleanplumb
+
+import android.net.Uri
+import io.mockk.coVerify
+import io.mockk.every
+import io.mockk.mockk
+import kotlinx.coroutines.test.runTest
+import mozilla.components.support.test.robolectric.testContext
+import mozilla.components.support.test.rule.MainCoroutineRule
+import mozilla.telemetry.glean.testing.GleanTestRule
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertNull
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mozilla.experiments.nimbus.NullVariables
+import org.mozilla.fenix.BuildConfig
+import org.mozilla.fenix.GleanMetrics.Messaging
+import org.mozilla.fenix.helpers.FenixRobolectricTestRunner
+import org.mozilla.fenix.nimbus.MessageData
+import org.mozilla.fenix.nimbus.StyleData
+import java.util.*
+
+@RunWith(FenixRobolectricTestRunner::class)
+class NimbusMessagingControllerTest {
+    private val storage: NimbusMessagingStorage = mockk(relaxed = true)
+
+    @get:Rule
+    val gleanTestRule = GleanTestRule(testContext)
+
+    private val coroutinesTestRule = MainCoroutineRule()
+    private val coroutineScope = coroutinesTestRule.scope
+
+    private val controller = NimbusMessagingController(storage) { 0L }
+
+    @Before
+    fun setup() {
+        NullVariables.instance.setContext(testContext)
+    }
+
+    @Test
+    fun `WHEN calling onMessageDismissed THEN record an event and updates metadata`() = coroutineScope.runTest {
```

Comment:
instead of 'an event' would it be useful to have the event type e.g. shown message event